### PR TITLE
feat: refactor favorite items

### PR DESCRIPTION
**Pull request by morganridel**
Based on branch `typeorm`

Related PRs:
* SDK: https://github.com/graasp/graasp-sdk/pull/164
* query-client: https://github.com/graasp/graasp-query-client/pull/275
* Builder: https://github.com/graasp/graasp-builder/pull/656

Favorite items used to be a field in the extra field of members. This PR is making it a real entity of the relational database model. This implies some migrations to handle the existing data and creating the new table.

Once everything is reviewed, this can be merged on typeorm, or this can wait for the refactor merge to be merged on main.

### CODE REVIEW

File: src/services/item/plugins/itemFavorite/schemas.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    favorite: {
+      type: 'object',
+      properties: {
+        id: {
+          $ref: 'http://graasp.org/#/definitions/uuid',
+        },
+        item: {
+          $ref: 'http://graasp.org/items/#/definitions/item',
+        },
+        createdAt: {},
```
Comment (spaenleh) [inline src/services/item/plugins/itemFavorite/schemas.ts:13]: Is this supposed to be a `string` since it is serialized ?

**Review by spaenleh — APPROVED** [review]
🌟 This looks good to me, but I let others chime in with their comments.

**Review by spaenleh — APPROVED** [review]

**Review by pyphilia — APPROVED** [review]
LGTM, I'm wondering if it would be necessary to add more care for the `down` operation, like adding back the favorites in the member and test that.

File: src/services/item/plugins/itemFavorite/services/favorite.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  async getOwn(actor: Actor, repositories: Repositories) {
+    const { itemFavoriteRepository } = repositories;
+
+    if (!actor) {
+      throw new UnauthorizedMember(actor);
+    }
+
+    return (await itemFavoriteRepository.getFavoriteForMember(actor.id)).filter(
+      async (f) =>
+        (await validatePermission(repositories, PermissionLevel.Read, actor, f.item)) !== null,
```
[the remark this answers is not in the document]
Comment (morganridel) [reply src/services/item/plugins/itemFavorite/services/favorite.ts:25]: You're right

File: src/migrations/1683637099103-add-favorites.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  }
+
+  public async down(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      'ALTER TABLE "item_favorite" DROP CONSTRAINT "FK_10ea93bde287762010695378f94"',
+    );
+    await queryRunner.query(
+      'ALTER TABLE "item_favorite" DROP CONSTRAINT "FK_a169d350392956511697f7e7d38"',
+    );
+    await queryRunner.query('DROP TABLE "item_favorite"');
```
[the remark this answers is not in the document]
Comment (morganridel) [reply src/migrations/1683637099103-add-favorites.ts:36]: discussed

**Pull request merged**